src/Framework/TestCase.php

```diff
@@ -1109,33 +1112,120 @@ public function runBare(): void
 
         $hasMetRequirements = false;
 
+        $emitter = Event\Registry::emitter();
+
         try {
             $this->checkRequirements();
             $hasMetRequirements = true;
 
             if ($this->inIsolation) {
+                $methodsCalledBeforeFirstTest = [];
+
                 foreach ($hookMethods['beforeClass'] as $method) {
                     $this->{$method}();
+
+                    $methodCalledBeforeFirstTest = new Event\Code\ClassMethod(
+                        static::class,
+                        $method
+                    );
+
+                    $emitter->testBeforeFirstTestMethodCalled(
+                        static::class,
+                        $methodCalledBeforeFirstTest
+                    );
+
+                    $methodsCalledBeforeFirstTest[] = $methodCalledBeforeFirstTest;
                 }
+
+                $emitter->testBeforeFirstTestMethodFinished(
+                    static::class,
+                    ...$methodsCalledBeforeFirstTest
+                );
             }
 
             $this->setDoesNotPerformAssertionsFromAnnotation();
 
+            $methodsCalledBeforeTest = [];
+
             foreach ($hookMethods['before'] as $method) {
                 $this->{$method}();
+
+                $methodCallBeforeTest = new Event\Code\ClassMethod(
+                    static::class,
+                    $method
+                );
+
+                $emitter->testBeforeTestMethodCalled(
+                    static::class,
+                    $methodCallBeforeTest
+                );
+
+                $methodsCalledBeforeTest[] = $methodCallBeforeTest;
             }
 
+            $emitter->testBeforeTestMethodFinished(
+                static::class,
+                ...$methodsCalledBeforeTest
+            );
+
+            $emitter->testSetUpFinished();
+
+            $methodsCalledPreCondition = [];
+
             foreach ($hookMethods['preCondition'] as $method) {
                 $this->{$method}();
+
+                $methodCalledPreCondition = new Event\Code\ClassMethod(
+                    static::class,
+                    $method
+                );
+
+                $emitter->testPreConditionCalled(
+                    static::class,
+                    $methodCalledPreCondition
+                );
+
+                $methodsCalledPreCondition[] = $methodCalledPreCondition;
             }
 
+            $this->assertPreConditions();
+
+            $emitter->testPreConditionFinished(
+                static::class,
+                ...$methodsCalledPreCondition
+            );
+
+            $emitter->testPrepared(new Event\Code\Test(
+                static::class,
+                $this->name
+            ));
+
             $this->testResult = $this->runTest();
+
             $this->verifyMockObjects();
```

Comment:
@sebastianbergmann @theseer 

Similarly, should we also

- add this method to `$methodsCalledPostCondition` _and_ 
- emit a `Test\PostConditionCalled` event for this method

here?